flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkResourceManager.java

```diff
@@ -78,6 +79,9 @@
 	/** The containers where a TaskManager is starting and we are waiting for it to register */
 	private final Map<ResourceID, YarnContainerInLaunch> containersInLaunch;
 
+	/** The container where a TaskManager has been started and is running in */
+	private final Map<ResourceID, Container> containersLaunched;
```

Comment:
I never said that it's yours but you've been much more involved in the design of this component than I was. Thus, I assumed that you know more than I do what the semantics of the `registeredWorkers` fields is. For example, does this field represent currently registered TMs at a JM or does this field represents the TMs which were registered at the RM by the JM?

I didn't know about it and couldn't deduce it from the code. Consequently, I chose the safest approach to leave it as it is and introduce a new container state in the `YarnFlinkResourceManager` implementation.

But if you think that clearing is just a bug and that the corresponding JavaDoc comment is outdated, then it might be a good option to change it.